aws/cdk/lib/cdk-pipeline-stack.ts

```diff
@@ -0,0 +1,285 @@
+import { Construct } from 'constructs';
+import { Secret } from "aws-cdk-lib/aws-secretsmanager";
+import * as codestarnotifications from "aws-cdk-lib/aws-codestarnotifications";
+import * as sns from "aws-cdk-lib/aws-sns";
+
+interface ViiteCdkStackProps extends cdk.StackProps {
+  environment: 'Dev' | 'QA';
+  pipelineName: string;
+  buildProjectName: string;
+  gitHubBranch: string;
+  artifactBucketName: string;
+  ecrRepositoryName: string;
+  securityGroupId: string;
+  vpcId: string;
+  ecsClusterName: string;
+  ecsServiceName: string;
+}
+
+export class ViiteCdkStack extends cdk.Stack {
+  constructor(scope: Construct, id: string, props: ViiteCdkStackProps) {
+    super(scope, id, props);
+
+    const { environment, pipelineName, buildProjectName, gitHubBranch, artifactBucketName, ecrRepositoryName, securityGroupId, vpcId, ecsClusterName, ecsServiceName } = props;
+
+    // Add a description for the stack
+    this.templateOptions.description = `Viite CI/CD Pipeline Stack for ${environment} environment including CodeBuild projects, CodePipeline, and associated resources.`;
+
+    // Import existing resources
+    const artifactBucket = s3.Bucket.fromBucketName(this, 'ExistingBucket', artifactBucketName);
+    const ecrRepository = ecr.Repository.fromRepositoryName(this, 'ExistingECRRepo', ecrRepositoryName);
+    const vpc = ec2.Vpc.fromLookup(this, 'ExistingVPC', {
+      vpcId: vpcId
+    });
+    const vpcSubnets = vpc.selectSubnets({
+      subnetType: ec2.SubnetType.PRIVATE_WITH_EGRESS
+    });
+    const securityGroup = ec2.SecurityGroup.fromSecurityGroupId(this, 'ExistingSecurityGroup', securityGroupId);
+
+    // Create SNS Topic for notifications
+    const notificationTopic = new sns.Topic(this, 'PipelineNotificationTopic', {
+      topicName: `${pipelineName}-notifications`,
+      displayName: `Viite ${environment} Pipeline Notifications for build failure and deployment success`
+    });
+
+    // CodeBuild Project
+    const buildProject = new codebuild.PipelineProject(this, 'BuildProject', {
+      projectName: buildProjectName,
+      environment: {
+        buildImage: codebuild.LinuxBuildImage.STANDARD_5_0,
+        computeType: codebuild.ComputeType.SMALL,
+        privileged: true,
+        environmentVariables: {
+          bonecpUsername: { value: '/dev/viite_test.db.username', type: codebuild.BuildEnvironmentVariableType.PARAMETER_STORE },
+          bonecpPassword: { value: '/dev/viite_test.db.password', type: codebuild.BuildEnvironmentVariableType.PARAMETER_STORE },
+          bonecpJdbcUrl: {
+            value: `/Viite/${environment}/bonecpJdbcUrl`,
+            type: codebuild.BuildEnvironmentVariableType.PARAMETER_STORE
+          },
+          vvhRestApiEndPoint: { value: 'vvhRestApiEndPoint', type: codebuild.BuildEnvironmentVariableType.PARAMETER_STORE },
+          vvhServiceHost: { value: 'vvhServiceHost', type: codebuild.BuildEnvironmentVariableType.PARAMETER_STORE },
```

Comment:
vvh:ta ei ole enää, Tarvitaanko näitä kahta? ^1, ^2
Niiden tilalla kgv. Onko vain nimet väärin, vai kokonaan turhia?